components/suggest/src/db.rs

```diff
@@ -139,12 +140,14 @@ impl<'a> SuggestDao<'a> {
                      SELECT s.id, k.rank, s.title, s.url, s.provider, k.confidence, k.keyword_suffix
                      FROM suggestions s
                      JOIN pocket_keywords k ON k.suggestion_id = s.id
-                     WHERE k.keyword_prefix = :keyword_prefix",
+                     WHERE k.keyword_prefix = :keyword_prefix
+                     LIMIT :suggestions_limit",
                     providers_to_sql_list(&query.providers),
```

Comment:
One other note for our future selves: unless it can use an indexed column, SQLite will build an in-memory B-Tree for sorting with `ORDER BY`. The cost can add up with a lot of rows, or with large rows, but I think it's totally okay in our case here: it makes results and tests deterministic, it sets the stage for ranking suggestions later, and, if we have more than a handful of suggestions to show, the awesomebar will run out of space long before sorting the suggestions becomes a bottleneck.